# Ciclo For 

In [1]:
x = [2,3,4,5]
prod = 1

In [2]:
for val in x:
    x = prod
    prod = prod * val
    print ("Moltiplicando "+str(val)+ " per " +str(x) +" otteniamo " 
           +str(prod)) 

Moltiplicando 2 per 1 otteniamo 2
Moltiplicando 3 per 2 otteniamo 6
Moltiplicando 4 per 6 otteniamo 24
Moltiplicando 5 per 24 otteniamo 120


Risultato:<br>
Moltiplicando 2 per 1 otteniamo 2<br>
Moltiplicando 3 per 2 otteniamo 6<br>
Moltiplicando 4 per 6 otteniamo 24<br>
Moltiplicando 5 per 24 otteniamo 120

 funzione Range

In [3]:
list(range(5))
# Risultato [0, 1, 2, 3, 4]

[0, 1, 2, 3, 4]

In [4]:
list(range(7,15))
# Risultato [7, 8, 9, 10, 11, 12, 13, 14]

[7, 8, 9, 10, 11, 12, 13, 14]

In [5]:
list(range(7,15,2))
# Risultato [7, 9, 11, 13]

[7, 9, 11, 13]

 Funzione range in for loops

In [6]:
item = ['banana', 'mela', 'arancia']

In [7]:
for val in range(len(item)):
    print ('Vorrei mangiare una',item[val])

Vorrei mangiare una banana
Vorrei mangiare una mela
Vorrei mangiare una arancia


Risultato:<br>
Vorrei mangiare una banana<br>
Vorrei mangiare una mela<br>
Vorrei mangiare una arancia    

 Ciclo For con blocco else

In [8]:
num = 1

In [9]:
for val in range(num,25,3):
    print(val)
else:
    print('Nessun elemento rimanente')

1
4
7
10
13
16
19
22
Nessun elemento rimanente


Risultato:<br>
1<br>
4<br>
7<br>
10<br>
13<br>
16<br>
19<br>
22<br>
Nessun elemento rimanente

In [10]:
età = 35
dict1 = {25: 'ragazzo',42: 'donna', 7: 'bimbo', 57:'signore'}

SyntaxError: invalid non-printable character U+00A0 (666986519.py, line 1)

In [ ]:
for val in dict1:
    if val == età:
        print (dict1[età])
        break
else:
    print("Età della persona non presente nel dizionario")

Risultato:<br>
Età della persona non presente nel dizionario

-----------------------------------------------------------

# While loops

In [ ]:
a = (5,7,98,24,47)
i = 0
somma = 0
while i < 5:
    print(a[i])
    somma = somma + a[i]
    i = i + 1
print ('la somma della tupla è pari a: '+str(somma))

Risultato:<br>
5<br>
7<br>
98<br>
24<br>
47<br>
la somma della tupla è pari a: 181

 Ciclo while con blocco else

In [ ]:
import random
count = 1

In [ ]:
while count < 8:
   guess = random.randint(1,8)    
   if count >= 1:
        count +=1
       
   if guess == count:
        print ("il numero random e count sono pari a "+str(guess))
        print ("Hai vinto!")
        
        break
   elif count == 8:
        print ("Hai perso")
        print ("dopo 8 iterazioni count è diverso da guess")
else:
    print ("Prova ancora")
        
# Risultato:
# Hai perso
# dopo 8 iterazioni count è diverso da guess
# Prova ancora    

 Istruzione break<br>
# Ciclo For 

In [ ]:
x = [2,3,4,5]
prod = 1

In [ ]:
for val in x:
    x = prod
    prod = prod * val
    if prod > 20:
        print ('X è pari a '+str(x) +' mentre prod è uguale a '+str(prod))
        break

Risultato<br>
X è pari a 6 mentre prod è uguale a 24

 break ciclo while

In [ ]:
a = (5,7,98,24,47)
i = 0
somma = 0
while i < 5:
    somma = somma + a[i]
    i = i + 1
    if i == 4:
       print ('Raggiunta la quarta iterazione del ciclo!'+
              ' La somma della tupla è pari a: '+str(somma))
       break
# Risultato:
# Raggiunta la quarta iterazione del ciclo! La somma della tupla è pari a: 134    
    
## Istruzione Continue
# Ciclo for

In [ ]:
vals = [1, 2, 3, 4, 5]
for i in vals:
    if i == 2:
        continue
    print(i**2)
    
# Risultato: 
# 1
# 9
# 16
# 25
 

Ciclo while

In [ ]:
vals = [1, 2, 3, 4, 5]
i = 1
while i <= 5:
    if i == 2:
        i=i+1
        continue
        
    print(i**2)
    i += 1
# Risutalto:
# 1
# 9
# 16
# 25    